hack/charts/cluster-api-operator/templates/infra.yaml

```diff
@@ -38,15 +38,15 @@ metadata:
   annotations:
     "helm.sh/hook": "post-install"
     "helm.sh/hook-weight": "2"
-{{- if or $infrastructureVersion $.Values.configSecret.name }}
+{{- if or $infrastructureVersion $.Values.configSecret.name $.Values.manager }}
 spec:
 {{- end }}
 {{- if $infrastructureVersion }}
   version: {{ $infrastructureVersion }}
 {{- end }}
 {{- if $.Values.manager }}
   manager:
-{{- if hasKey $.Values.manager.featureGates $infrastructureName }}
+{{- if and (kindIs "map" $.Values.manager.featureGates) (hasKey $.Values.manager.featureGates $infrastructureName) }}
```

Comment:
This fixes a type check error if $.Values.manager.featureGates is not specified.